openlibrary/plugins/upstream/checkins.py

```diff
@@ -118,6 +118,9 @@ def validate_data(self, data):
         if 'event_type' not in data:
             return False
 
+        if not data.get('event_type') in BookshelfEvent:
```

Comment:
@cclauss, is this a non-standard way to check if a value is in an enum?  Is there a better way of doing this than defining a metaclass for the `BookshelfEvent` enum?

Reply:
All Enums are already instances of EnumMeta (renamed EnumType in Py3.11) so I think the best solution is to slightly rename `.has_key()` to `.has_key_str()`...
```python
class BookshelfEvent(IntEnum):
    START = 1
    UPDATE = 2
    FINISH = 3

    @classmethod
    def has_key_str(cls, key_str: str) -> bool:
        return key_str in cls.__members__

BookshelfEvent.has_key_str("FINISH")  # --> True
```

Reply:
I'm looking for the values, not the keys (`event_type` should be 1, 2, or 3).
Should I create a `.has_value(value : int)` method instead of overriding `__contains__`?

Reply:
Yes because `.__contains__()` is about keys, not about values...
```
@classmethod
    def has_value(cls, value: int) -> bool:
        return value in (item.value for item in BookshelfEvent.__members__.values())
```